File: cpp/src/ev2/ut6_pooavanzadaestructuras/u01herenciapolimorfismo/practicas/madridcesjuanpablosegundoalumnos/alumnos_resuelto.cpp
#include <iostream>
#include <vector>
#include <string>
using namespace std;
class Alumno {
public:
  int id; string nombre, apellido; string telefono; string correo; bool hasTel=false, hasCorreo=false;
  double calificacion=0; bool matriculado=true;
  Alumno(int id, string n, string a): id(id), nombre(n), apellido(a) {}
  void mostrarDatos(){
    cout<<"id = "<<id<<"\nnombre = "<<nombre<<"\napellido = "<<apellido<<"\n";
    cout<<"telefono = "<<(hasTel?telefono:"dato no disponible")<<"\n";
    cout<<"correo = "<<(hasCorreo?correo:"dato no disponible")<<"\n";
    cout<<"calificacion = "<<calificacion<<"\nmatriculado = "<<matriculado<<"\n";
  }
  void desmatricular(){ matriculado=false; }
  void calificar(double n){ calificacion=n; }
};
int main(){ vector<Alumno> alumnos; Alumno a(1,"Ada","Lovelace"); a.hasCorreo=true; a.correo="ada@mail.com"; a.calificar(9.5); a.mostrarDatos(); return 0; }

File: js/ev2/ut5-poo-colecciones/03-objetos-y-clases/practicas/madrid-cesjuanpablosegundo-alumnos/alumnos-resuelto.js
class Alumno {
  constructor(id, nombre, apellido, telefono=null, correo=null) {
    this.id=id; this.nombre=nombre; this.apellido=apellido; this.telefono=telefono; this.correo=correo;
    this.calificacion=0; this.matriculado=true;
  }
  mostrarDatos(){
    console.log(`id = ${this.id}`); console.log(`nombre = ${this.nombre}`);
    console.log(`telefono = ${this.telefono ?? "dato no disponible"}`);
    console.log(`correo = ${this.correo ?? "dato no disponible"}`);
    console.log(`calificacion = ${this.calificacion}`); console.log(`matriculado = ${this.matriculado}`);
  }
  desmatricular(){ this.matriculado=false; }
  calificar(n){ this.calificacion=n; }
}
const alumnos = [new Alumno(1,"Ada","Lovelace",null,"ada@mail.com")];
alumnos[0].calificar(9.5); alumnos[0].mostrarDatos();
